### PR TITLE
[12.0] IMP base_model_restrict_update: readonly users can create wizards

### PR DESCRIPTION
**Pull request by eLBati**
This is useful for reporting wizards typically

### CODE REVIEW

**Review by yostashiro — APPROVED** [review]
Code review + functional test. LGTM 👍

**Comment by eLBati** [comment]
Hi @yostashiro , please note I just updated this PR with https://github.com/OCA/server-tools/pull/2287/commits/263144692a357360503bf3c82deb100478828b11

**Comment by yostashiro** [comment]
> Hi @yostashiro , please note I just updated this PR with [2631446](https://github.com/OCA/server-tools/commit/263144692a357360503bf3c82deb100478828b11)

@eLBati If I'm not mistaken, I think this change opens more possibility to bypass the ACL check in an unexpected way for non-readonly users.

To make the new settings applicable only to the readonly users, how about updating this part
https://github.com/OCA/server-tools/blob/311142a8aab02bb9f6a2fee9fa0d01d539bacaee/base_model_restrict_update/models/ir_model_access.py#L32-L37
to something like this:
```py
    @api.model
    def _test_readonly(self, model):
        if not self.env.user.is_readonly_user:
            return True
        if model not in exclude_models self._readonly_exclude_models():
            return True
        return False
```

I also think the new changes deserve updates on tests and readme.

**Comment by eLBati** [comment]
```
        if not self.env.user.is_readonly_user:
            return True
```
:thinking: why should `_test_readonly` return True if user is not readonly?

> I also think the new changes deserve updates on tests and readme

Right, I proceed

**Comment by eLBati** [comment]
> Right, I proceed

@yostashiro done

**Comment by yostashiro** [comment]
> 🤔 why should `_test_readonly` return True if user is not readonly?

@eLBati You are right. I made a mistake.

I think, however, it's worth adjusting the method to
```py
    @api.model
    def _test_readonly(self, model):
        if not self.env.user.is_readonly_user:
            return False
        if model in exclude_models self._readonly_exclude_models():
            return False
        return True
```
to skip unnecessary searches for non-readonly users now that you are adding more searches in _readonly_exclude_models().

**Comment by eLBati** [comment]
@yostashiro ok, changed, thanks

File: base_model_restrict_update/models/ir_model_access.py [hunk 1]
```diff
@@ -48,8 +49,13 @@ def _test_restrict_update(self, model):
...
     def _readonly_exclude_models(self):
-        """ Models updtate/create by system, and should be excluded from checking """
-        return self.sudo().search([
+        transient_models = self.env["ir.model"].sudo().search(
+            [("transient", "=", True)]).mapped("model")
+        skipped_models = self.env["ir.model"].sudo().search([
+            ("skip_check_for_readonly_users", "=", True)
+        ]).mapped("model")
+        # Models updtate/create by system, and should be excluded from checking
+        return skipped_models + transient_models + self.sudo().search([
```
Comment (yostashiro) [inline base_model_restrict_update/models/ir_model_access.py]:
```suggestion
        skipped_models = self.env["ir.model"].sudo().search(
            ["|", ("transient", "=", True), ("skip_check_for_readonly_users", "=", True)]
        ).mapped("model")
        # Models updtate/create by system, and should be excluded from checking
        return skipped_models + self.sudo().search([
```
Let's combine the searches into one. ;)

Comment (eLBati) [reply base_model_restrict_update/models/ir_model_access.py]: committed, thanks

**Review by yostashiro — APPROVED** [review]
LGTM 👍

**Review by newtratip — APPROVED** [review]
LGTM :+1:

**Pull request merged**